packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -90,6 +90,10 @@ library LibProving {
         if (tid == 0) {
             tid = blk.nextTransitionId;
 
+            // Unchecked is safe:
+            // - Not realistic 65K different fork choice per block will be
+            // proven
```

Comment:
65K is not realistic, but on the other hand it is theoretically possible:
```
30,000,000 (Ethereum block gas limit now) / 40,000 (low ball fork choice creation cost) = 750 fork choices/block.
65,000 / 750 = 86 Ethereum blocks
```
So it only takes ~86 current Ethereum blocks to wrap this around, which isn't that much and this could easily be done. So this indirectly depends on how expensive it is to create proofs and the cost onchain to validate them. I feel this is a bit of a shaky foundation for something so critical.

Reply:
I would make it uint32 to prevent any possible problems.

Reply:
Addressed in this PR: https://github.com/taikoxyz/taiko-mono/pull/14620